client/packages/react/src/next-ssr/HydrationStreamProvider.tsx

```diff
@@ -0,0 +1,198 @@
+'use client';
+
+// This allows us to send serialized json from the server to the client using 'useServerInsertedHTML'
```

Comment:
Maybe one thing that would be nice in this comment: 

Would start with "what is the problem". i.e why do we need to send serialized json from the server to the client? 

What is useServerInsertedHTML?